examples/fungible-token/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+use frc42_dispatch::method_hash;
+use fvm_actors_runtime::runtime::{ActorCode, Runtime};
+use fvm_actors_runtime::{actor_error, cbor, ActorDowncast, ActorError, INIT_ACTOR_ADDR};
+use fvm_ipld_blockstore::Blockstore;
+use fvm_ipld_encoding::RawBytes;
+use fvm_shared::address::Address;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::error::ExitCode;
+use fvm_shared::{MethodNum, METHOD_CONSTRUCTOR};
+use ipc_atomic_execution::{AtomicExecID, AtomicInputID};
+use ipc_gateway::{ApplyMsgParams, CrossMsg, IPCAddress, StorableMsg};
+use num_derive::FromPrimitive;
+use num_traits::FromPrimitive;
+use serde_tuple::{Deserialize_tuple, Serialize_tuple};
+use state::State;
+use std::collections::{HashMap, HashSet};
+
+mod state;
+
+fvm_actors_runtime::wasm_trampoline!(Actor);
+
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    Name = method_hash!("Name"),
+    Symbol = method_hash!("Symbol"),
+    TotalSupply = method_hash!("TotalSupply"),
+    Balance = method_hash!("Balance"),
+    Transfer = method_hash!("Transfer"),
+    InitAtomicTransfer = method_hash!("InitAtomicTransfer"),
+    PrepareAtomicTransfer = method_hash!("PrepareAtomicTransfer"),
+    CancelAtomicTransfer = method_hash!("CancelAtomicTransfer"),
+    CommitAtomicTransfer = method_hash!("CommitAtomicTransfer"),
+    AbortAtomicTransfer = method_hash!("AbortAtomicTransfer"),
+    RollbackAtomicTransfer = method_hash!("RollbackAtomicTransfer"),
+}
+
+struct Actor;
+
+#[derive(Clone, Debug, Serialize_tuple, Deserialize_tuple)]
+pub struct ConstructorParams {
+    pub ipc_gateway: Address,
+    pub ipc_address: IPCAddress,
+    pub name: String,
+    pub symbol: String,
+    pub balances: HashMap<Address, TokenAmount>,
```

Comment:
Why do we pass the `balances` in the constructor params? Wouldn't it be better to initialize this in the constructor?

Reply:
I believe [this](https://github.com/consensus-shipyard/ipc-atomic-execution/pull/6#discussion_r1061547726) answers your question.

Reply:
It does, thanks!